### DISCUSSION TITLE
Is it possible to manually via query decide how will be parameters combined?

### DISCUSSION THREAD

**Discussion by Cubody** [enisn/AutoFilterer#60] [Q&A]
@enisn Heya! I was thinking about complex query. For example I want filter like a == 3 && (b ==4 || c == 2).
Is it possible to do such filter via query?

**Comment by enisn** [comment]
I had a markup goal that covers this situation but it's not implemented yet: https://github.com/enisn/AutoFilterer/issues/18
The following markup can be used to achieve your requirement.
```csharp
[CompareTo("A")]
[And]
[CompareTo("B", "C", CombineWith = CombineType.Or)]
```

Unfortunately it's an **unimplemented** feature. So you can create your own attribute for your custom requirements or just you can write linq for your custom requirement.


You can follow this document to create custom attributes: https://enisn-projects.io/docs/en/AutoFilterer/latest/examples/Custom-Expression-for-Property

In your case following implementation will work:
```csharp
  public class MyCustomSearchAttribute : FilteringOptionsBaseAttribute
  {
      public override Expression BuildExpression(Expression expressionBody, PropertyInfo targetProperty, PropertyInfo filterProperty, object value)
      {
          // Also means:  x => x.A == "someValue" && (x.B == "someValue" || x.C == "somveValue")
          return Expression.And(
                  Expression.Equal(
                      Expression.Property(expressionBody, "A"),
                      Expression.Constant(value)
                      ),
                  Expression.Or(
                      Expression.Equal(
                          Expression.Property(expressionBody, "B"),
                          Expression.Constant(value)
                          ),
                      Expression.Equal(
                          Expression.Property(expressionBody, "C"),
                          Expression.Constant(value)
                          )
                      )
              );
      }
  }
```

Then you can use that attribute in your Filter Object

```csharp
public class MyFilter : FilterBase
{
    [MyCustomSearch]
    public int Filter { get; set; }
}
```

**Reply by enisn** [reply]
I see, I misunderstood your requirement. You mean, you'll pass even `Or`, `And` operator from query, right?

**Reply by Cubody** [reply]
> I see, I misunderstood your requirement. You mean, you'll pass even `Or`, `And` operator from query, right?

Yes, I meant passing it from query.
